sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonIUD.java

```diff
@@ -0,0 +1,275 @@
+      if (this.updateColumnToValueMapping.containsKey(path.getKey())) {
+        updateExecution(path.getKey());
+      }
+    }
+    this.filterColumnToValueMapping.clear();
+    this.updateColumnToValueMapping.clear();
+  }
+
+  private void updateExecution(String path)
+      throws IOException, InterruptedException, InvalidLoadOptionException {
+    Expression filterExpression = getExpression(path);
+    update(path, filterExpression, this.updateColumnToValueMapping.get(path));
+  }
+
+  private void deleteExecution(String path) throws IOException, InterruptedException {
+    Expression filterExpression = getExpression(path);
+    delete(path, filterExpression);
+  }
+
+  private void prepareUpdate(String path, String column, String value, String updColumn,
+      String updValue) {
+    prepareDelete(path, column, value);
+    updColumn = updColumn.toLowerCase().trim();
+    if (this.updateColumnToValueMapping.containsKey(path)) {
+      this.updateColumnToValueMapping.get(path).put(updColumn, updValue);
+    } else {
+      Map<String, String> columnToValue = new HashMap<>();
+      columnToValue.put(updColumn, updValue);
+      this.updateColumnToValueMapping.put(path, columnToValue);
+    }
+  }
+
+  private void prepareDelete(String path, String column, String value) {
+    column = column.toLowerCase().trim();
+    if (this.filterColumnToValueMapping.containsKey(path)) {
+      this.filterColumnToValueMapping.get(path).put(column, value);
+    } else {
+      Map<String, String> columnToValueMapping = new HashMap<>();
+      columnToValueMapping.put(column, value);
+      this.filterColumnToValueMapping.put(path, columnToValueMapping);
+    }
+  }
+
+  private Expression getExpression(String path) throws IOException {
+    File[] indexFiles = getCarbonIndexFile(path);
+    Schema schema = CarbonSchemaReader.readSchema(indexFiles[0].getAbsolutePath()).asOriginOrder();
+    Field[] fields = schema.getFields();
+    List<EqualToExpression> listOfExpressions = new ArrayList<>();
+    for (Map.Entry<String, String> column : this.filterColumnToValueMapping.get(path).entrySet()) {
+      DataType dataType = getColumnDataType(fields, column.getKey());
+      listOfExpressions.add(new EqualToExpression(new ColumnExpression(column.getKey(), dataType),
+          new LiteralExpression(column.getValue(), dataType)));
+    }
+    Expression filterExpression = null;
+    if (listOfExpressions.size() > 0) {
+      filterExpression = listOfExpressions.get(0);
+    }
+    for (int i = 1; i < listOfExpressions.size(); i++) {
+      filterExpression = new AndExpression(filterExpression, listOfExpressions.get(i));
+    }
+    return filterExpression;
+  }
+
+  private int getColumnIndex(Field[] fields, String column) {
+    int index = -1;
+    for (Field field : fields) {
+      if (field.getFieldName().equals(column)) {
+        index = field.getSchemaOrdinal();
+        break;
+      }
+    }
+    if (index == -1) {
```

Comment:
Please  check  complex  data type  for it：
  array，  struct， map

Reply:
@Karan980 Please check this and add test cases for all the missing types as well

Reply:
Done